### PR TITLE
Update govuk-frontend from V2 to V3

### PR DESCRIPTION
**Pull request by domoscargin**
Part of https://trello.com/c/5FhjhxMd/139-3-migrate-user-frontend-to-govuk-frontend-v3

### CODE REVIEW

**Comment by lfdebrux** [comment]
I think it might make our lives easier later down the line if we use `govuk/` as the path for govuk-frontend source for both version 3 and version 2; I've put together a PR at #151 that does this, see what you think?

File: src/digitalmarketplace/components/alert/_alert.scss [hunk 1]
```diff
@@ -1,10 +1,9 @@
-@import "../../../govuk-frontend/tools/all";
-@import "../../../govuk-frontend/helpers/all";
+@import "../../../govuk-frontend/govuk/tools/all";
+@import "../../../govuk-frontend/govuk/helpers/all";
 
 @include govuk-exports("digitalmarketplace/component/alert") {
 
     .dm-alert {
-        @include govuk-focusable;
```
Comment (lfdebrux) [inline src/digitalmarketplace/components/alert/_alert.scss:7]: Do we need to do anything to replace `govuk-focusable`?

**Comment by lfdebrux** [comment]
Thinking about it, I think we will need to merge these changes to a long-running branch and not than the default one, so if we can put as many of the changes as possible into a PR that we can merge into main that would be awesome.

File: src/digitalmarketplace/components/alert/_alert.scss (hunk 1 above)
Comment (domoscargin) [reply src/digitalmarketplace/components/alert/_alert.scss:7]: > Thinking about it, I think we will need to merge these changes to a long-running branch and not than the default one, so if we can put as many of the changes as possible into a PR that we can merge into main that would be awesome.
> 
> I've already split the `npm audit` bumps into a separate PR at #150

I think the rest of this is now govuk-v3 dependent. I can rip out all the path changes so this is more easily merged with #151 though.

Potentially could also push the `data-module="govuk-button"` change to main, though that at least has potential to do something weird.

**Comment by domoscargin** [comment]
> Thinking about it, I think we will need to merge these changes to a long-running branch and not than the default one, so if we can put as many of the changes as possible into a PR that we can merge into main that would be awesome.
> 
> I've already split the `npm audit` bumps into a separate PR at #150

I think the rest of this is now govuk-v3 dependent. I can rip out all the path changes so this is more easily merged with #151 though.

Potentially could also push the `data-module="govuk-button"` change to main, though that at least has potential to do something weird.

**Comment by domoscargin** [comment]
Changed the base branch to govuk-frontend-3, which at the mo is just a straight clone of main.

**Comment by lfdebrux** [comment]
Do you think we can backport the changes to the banner and alert focus styles?

**Comment by domoscargin** [comment]
> Do you think we can backport the changes to the banner and alert focus styles to the main branch?

Looks like we could do most of it except for the `govuk-focused-text` which was introduced in 3.0.0. I'll keep that in here.

**Pull request merged**